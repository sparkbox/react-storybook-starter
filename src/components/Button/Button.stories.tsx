import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { Button } from '.';

export default {
  title: 'Components/Button',
};

export const WithText = () => <Button text="I'm a Button!" />;
