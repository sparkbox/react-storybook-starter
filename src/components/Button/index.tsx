import React, { ReactElement } from 'react';

type Props = {
  text: string;
};

export function Button({ text }: Props): ReactElement {
  return <button type="button">{text}</button>;
}
