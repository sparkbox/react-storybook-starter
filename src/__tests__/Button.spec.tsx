import * as React from "react";
import { shallow, render } from "enzyme";
import toJson from 'enzyme-to-json';

import Button from "../components/Button";

describe('the Button component', () => {
  it('matches the Snapshot', () => {
    const result = render(<Button text="Test Button" />);
    expect(toJson(result)).toMatchSnapshot();
  });

  it('renders a button element', () => {
      const result = shallow(<Button text="Test Button" />).contains(
        <button type="button">Test Button</button>
      );
      expect(result).toBeTruthy();
  });
});
