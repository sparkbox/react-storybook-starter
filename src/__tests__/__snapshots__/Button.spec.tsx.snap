// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Button component matches the Snapshot 1`] = `
<button
  type="button"
>
  Test Button
</button>
`;
